lib/inspec/resources/users.rb

```diff
@@ -314,6 +336,59 @@ def credentials
 
       @cred_cache = @user_provider.credentials(@username) unless @user_provider.nil?
     end
+
+    # helper method for has_authorized_key matcher
+    # checks for key is present as part of authorized keys on the server
+    def authorized_key(key)
+      # cat is used in unix system to display content of file; similarly type is used for windows
+      inspec.os.windows? ? bin = "type" : bin = "cat"
+
+      # auth_path gets assigned with the valid path for authorized_keys
+      auth_path = ""
+
+      # possible paths where authorized_keys are stored
+      %w{~/.ssh/authorized_keys ~/.ssh/authorized_keys2}.each do |path|
+        if inspec.command("#{bin} #{path}").exit_status == 0
+          auth_path = path
+          break
+        end
+      end
+
+      # if auth_path is empty, no valid path was found, hence raise exception
+      raise Inspec::Exceptions::ResourceSkipped, "Can't find any valid path for authorized_keys" if auth_path.empty?
+
+      # authorized_keys are obtained in the standard output;
+      # split keys on newline if more than one keys are part of authorized_keys
+      auth_keys = inspec.command("#{bin} #{auth_path}").stdout.split("\n").map(&:strip)
+
+      # check if the input key is part of the auth_keys
+      auth_keys.include?(key)
+    end
+
+    # Helper method for encrypted_password property
+    def encrypted_password_info
+      # check if getent is available on the system
+      bin = find_getent_utility
+
+      # fetch details of the passwd file for the current user using getent
+      cmd = inspec.command("#{bin} shadow #{@username}")
+
+      raise Inspec::Exceptions::ResourceSkipped, "Cannot view encrypted_password information" if cmd.exit_status != 0
```

Comment:
Does all non-zero exit status refer to not having permission to view encrypted_password information? Is it possible to view the exact error message to the User?

Reply:
Raised different message appending stderr